service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/DiagnosticReportMapperTest.java

```diff
@@ -201,6 +206,52 @@ void When_DiagnosticReport_With_RedactedFilingComment_Expect_ConfidentialityCode
         assertThat(actualXml).containsIgnoringWhitespaces(NOPAT_HL7_CONFIDENTIALITY_CODE);
     }
 
+    @Test
+    void When_DiagnosticReport_With_ObservationEffectiveDateTimeTypeAndCommentNote_Expect_NarrativeStatementConfidentialityCodePresent() {
+        final String diagnosticReportFileName = "diagnostic-report-with-effective-datetime-filing-comment.json";
+        final DiagnosticReport diagnosticReport = getDiagnosticReportResourceFromJson(diagnosticReportFileName);
+        final Bundle bundle = getBundleResourceFromJson(INPUT_JSON_BUNDLE_WITH_FILING_COMMENTS);
+        final Observation observation = (Observation) bundle.getEntry().get(2).getResource();
+        final InputBundle inputBundle = new InputBundle(bundle);
+        final List<String> expectedXPaths = Collections.singletonList(
+            "/component/CompoundStatement/component[1]/NarrativeStatement/" + getNopatConfidentialityCodeXpathSegment()
+        );
+
+        when(confidentialityService.generateConfidentialityCode(observation))
+            .thenReturn(Optional.of(NOPAT_HL7_CONFIDENTIALITY_CODE));
+        when(messageContext.getInputBundleHolder())
+            .thenReturn(inputBundle);
+
+        diagnosticReport.setStatus(null);
+
+        final String actualXml = mapper.mapDiagnosticReportToCompoundStatement(diagnosticReport);
+
+        assertThatXml(actualXml).containsAllXPaths(expectedXPaths);
+    }
+
+    @Test
+    void When_DiagnosticReport_With_ObservationEffectivePeriodAndCommentNote_Expect_NarrativeStatementConfidentialityCodePresent() {
```

Comment:
I think When_DiagnosticReport_With_ObservationEffectivePeriodAndCommentNote_Expect_NarrativeStatementConfidentialityCodePresent() and When_DiagnosticReport_With_ObservationEffectiveDateTimeTypeAndCommentNote_Expect_NarrativeStatementConfidentialityCodePresent() tests are exactly identical except the names, so maybe adjusting the name for one of the tests should be OK in order to remove the second duplicate test.

Reply:
Great spot, this certainly _should not_ be the case!

Reply:
Okay, I've pushed up a fix.

Note that the lines:

`final Observation observation = (Observation) bundle.getEntry().get(3).getResource();`
`final String diagnosticReportFileName = "diagnostic-report-with-effective-period-filing-comment.json";`

Do differ!